pkg/imgpkg/imagetar/tar_reader.go

```diff
@@ -28,6 +31,98 @@ func (r TarReader) Read() ([]imagedesc.ImageOrIndex, error) {
 	return imagedesc.NewDescribedReader(ids, file).Read(), nil
 }
 
+// GetLayersPresentInFile retrieves all the layers that are present in a tar file
+func (r TarReader) GetLayersPresentInFile() ([]v1.Layer, error) {
+	var result []v1.Layer
+	allImages, err := r.Read()
+	if err != nil {
+		return nil, err
+	}
+	for _, image := range allImages {
+		if image.Image != nil {
+			img := *image.Image
+			layers, err := r.getPresentLayersForImage(img)
+			if err != nil {
+				return nil, fmt.Errorf("Processing Image %s: %s", image.OrigRef, err)
+			}
+			result = append(result, layers...)
+		} else if image.Index != nil {
+			idx := *image.Index
+			layers, err := r.getPresentLayersForIndex(idx)
+			if err != nil {
+				return nil, fmt.Errorf("Processing Index %s: %s", image.OrigRef, err)
+			}
+			result = append(result, layers...)
+		}
+	}
+
+	return result, nil
+}
+
+func (r TarReader) getPresentLayersForImage(img v1.Image) ([]v1.Layer, error) {
+	var result []v1.Layer
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve layers: %s", err)
+	}
+
+	for _, layer := range layers {
+		h, err := layer.Digest()
+		if err != nil {
+			return nil, fmt.Errorf("Unable to get digest from layer: %s", err)
+		}
+		r, err := layer.Compressed()
+		if err != nil {
+			continue
+		}
+
+		size, err := layer.Size()
+		if err != nil {
+			return nil, err
+		}
+		closer, err := verify.ReadCloser(r, size, h)
+		if err != nil {
+			return nil, err
+		}
+
+		output := make([]byte, size)
+		n, err := closer.Read(output)
+		if err != nil {
```

Comment:
if err is not nil, then i dont think we should be considering this as a present layer.